pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -43,30 +43,23 @@ type K8sOrchestrator struct {
 }
 
 // Newk8sOrchestrator instantiates K8sOrchestrator object and returns this object
-func Newk8sOrchestrator(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor) (*K8sOrchestrator, error) {
+func Newk8sOrchestrator(ctx context.Context, featureStatesConfigMapInfo config.FeatureStatesConfigInfo) (*K8sOrchestrator, error) {
 	var coInstanceErr error
 	onceFork8sOrchestratorInstance.Do(func() {
+		log := logger.GetLogger(ctx)
 		log.Info("Initializing k8sOrchestratorInstance")
 		k8sOrchestratorInstance = &K8sOrchestrator{}
 		k8sOrchestratorInstance.featureStates = make(map[string]string)
-		log := logger.GetLogger(ctx)
 		k8sClient, coInstanceErr := k8s.NewClient(ctx)
 		if coInstanceErr != nil {
 			log.Errorf("Creating Kubernetes client failed. Err: %v", coInstanceErr)
 			return
 		}
-		var csiNamespace string
-		switch clusterFlavor {
-		case cnstypes.CnsClusterFlavorVanilla:
-			csiNamespace = common.CSINamespaceVanillaK8S
-		case cnstypes.CnsClusterFlavorWorkload:
-			csiNamespace = common.CSINamespaceWorkload
-		case cnstypes.CnsClusterFlavorGuest:
-			csiNamespace = common.CSINamespaceTkgCluster
-		}
-		fssConfigMap, err := k8sClient.CoreV1().ConfigMaps(csiNamespace).Get(ctx, common.CSIFeatureStatesConfigMapName, metav1.GetOptions{})
+		featureStatesConfigMapName = featureStatesConfigMapInfo.Name
+		featureStatesConfigMapNamespace = featureStatesConfigMapInfo.Namespace
+		fssConfigMap, err := k8sClient.CoreV1().ConfigMaps(featureStatesConfigMapNamespace).Get(ctx, featureStatesConfigMapName, metav1.GetOptions{})
 		if err != nil {
-			errMsg := fmt.Errorf("failed to fetch configmap %s. Setting the feature states to default values: %v. Error: %v", common.CSIFeatureStatesConfigMapName, k8sOrchestratorInstance.featureStates, err)
+			errMsg := fmt.Errorf("failed to fetch configmap %s from namespace %s. Setting the feature states to default values: %v. Error: %v", featureStatesConfigMapName, featureStatesConfigMapNamespace, k8sOrchestratorInstance.featureStates, err)
 			log.Debug(errMsg)
```

Comment:
remove- `errMsg`.

log.Debug(errMsg) => log.Errorf("failed to fetch configmap %s from namespace %s. Setting the feature states to default values: %v. Error: %v", featureStatesConfigMapName, featureStatesConfigMapNamespace, k8sOrchestratorInstance.featureStates, err)

Reply:
Done